src/staticdata.c

```diff
@@ -1436,23 +1463,32 @@ void gc_sweep_sysimg(void)
 static void _jl_write_value(jl_serializer_state *s, jl_value_t *v)
 {
     if (v == NULL) {
-        write_uint32(s->s, 0);
+        if (sizeof(reloc_t) <= 4) {
```

Comment:
This could be moved to a `write_reloc` function.

Reply:
yeah, it just seemed few enough places it should be needed that it wasn't worthwhile